### PR TITLE
Sprint 47

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 47

- 5537 Court: Disclosure Cases
- 6025 Data Migration: Migrate legacy cases that are calendared for trial
- 6070 Docket Clerks: Sealed Addresses
- [DevEx 746:](https://trello.com/c/MleaEzqE/746-repeatedly-migrating-a-case-that-includes-a-trial-session-id-results-in-the-case-being-listed-in-the-trial-session-multiple-time) Duplicate Cases on Trial Session Calendar
- [DevEx](https://trello.com/c/fN0ACplR/674-document-rules-we-follow-in-our-code): Document rules we follow in our code
- [DevEx](https://trello.com/c/Jsi9AijE/210-break-casedetailhelper-up-into-smaller-computeds) Break caseDetailHelper up into smaller computeds
- [DevEx:](https://trello.com/c/FQwEWO6v/595-dry-up-usage-of-testpdfdocbytes-throughout-application) DRY up usage of testPdfDocBytes
- DevEx: force exit when unit tests are finished running

### CODE REVIEW

File: Dockerfile-CI [hunk 1]
```diff
@@ -10,6 +10,11 @@ RUN apt-get -o Acquire::Check-Valid-Until=false update && \
 
 ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64
 
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+RUN unzip awscliv2.zip
+RUN ./aws/install
+RUN rm awscliv2.zip
+
```
Comment (julialeague) [inline Dockerfile-CI:17]: @ericsorenson Should we get the fix for the Java issues in with this change too? Otherwise it may break for some folks during the interim. Didn't you downgrade the dynamodb version?

File: docs/DEVELOPER_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Developer Guidelines
+
+* When using entity constants in files within the `/shared` or `/web-api` directory, directly import them at the top of the file. When using them in files within the `/web-client` directory, retrieve them using `applicationContext.getConstants()`. 
+
+* Don't use a computed value in another computed or set state within a computed.
```
Comment (julialeague) [inline docs/DEVELOPER_GUIDELINES.md]: @ericsorenson Any chance you could include an example here? The context for this escapes me.

File: shared/src/business/entities/contacts/ContactFactory.js [hunk 3]
```diff
@@ -454,6 +455,8 @@ ContactFactory.createContactFactory = ({
       this.countryType = rawContact.countryType;
       this.email = rawContact.email;
       this.inCareOf = rawContact.inCareOf;
+      this.isAddressSealed = rawContact.isAddressSealed || false;
```
Comment (julialeague) [inline shared/src/business/entities/contacts/ContactFactory.js:458]: @ericsorenson Why is the field required when it has a default value?

**Comment by julialeague** [comment]
We use this list when performing a code review to ensure that all tasks have been completed.

- [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
- [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
		- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
	- [x] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by julialeague** [comment]
@ericsorenson [SonarCloud](https://sonarcloud.io/organizations/flexion-github/projects) is showing a failed quality gate on the shared directory for duplicate code. Would you add a ticket to the backlog to address this?

**Comment by julialeague** [comment]
During local testing, I noticed that as a **petitioner**, clicking on my Petition document link on my docket record doesn't produce a result in Safari 13.1.2. There's no pop-up blocker notification or anything. Expected behavior: when I click on the Petition document link on my docket record, my petition document is opened in the browser.

**Comment by julialeague** [comment]
Is jQuery being used on this project? [Snyk shows a known vulnerability for XSS](https://snyk.io/vuln/npm:jquery?lh=3.4.1&utm_source=lighthouse&utm_medium=ref&utm_campaign=audit) on the version of jQuery installed. Any chance we could upgrade to the latest jQuery?

**Comment by ericsorenson** [comment]
@julialeague this project doesn't use jQuery.

**Comment by ericsorenson** [comment]
@julialeague regarding the SonarCloud quality gate, this caught the codebase in the middle of our Document / DocketRecord refactoring task. It's still being worked on, and will be sorted in the next PR.

**Comment by ericsorenson** [comment]
@julialeague regarding the Petition document link, Safari has a default popup blocker. The icon is on the right side of the address bar.
Screen Shot 2020-09-01 at 3 58 59 PM

**Comment by julialeague** [comment]
> @julialeague this project doesn't use jQuery.

That's strange! Lighthouse reported the outdated jQuery version being run. Not sure why. I'll let it go!

File: Dockerfile-CI (hunk 1 above)
Comment (ericsorenson) [reply Dockerfile-CI:17]: We didn't make any Java version changes. The DynamoDB version downgrade is in the next PR. As long as the cached version isn't deleted, running locally won't be an issue.

Comment (julialeague) [reply Dockerfile-CI:17]: > @julialeague regarding the Petition document link, Safari has a default popup blocker. The icon is on the right side of the address bar.

After clearing my cache, I have the same behavior as you. No worries!

**Comment by julialeague** [comment]
> @julialeague regarding the Petition document link, Safari has a default popup blocker. The icon is on the right side of the address bar.

After clearing my cache, I have the same behavior as you. No worries!

File: docs/DEVELOPER_GUIDELINES.md (hunk 2 above)
Comment (ericsorenson) [reply docs/DEVELOPER_GUIDELINES.md]: Updated!

**Review by julialeague — APPROVED** [review]
@mmarcotte, I recommend you accept this PR into the court environment. Thanks!
